nautobot/docs/development/core/model-checklist.md

```diff
@@ -0,0 +1,139 @@
+# Model Development Checklist
+
+A general best-practices checklist to follow when adding a net-new data model in Nautobot or adding new fields to an existing model.
+
+## Bootstrapping a new Model
```

Comment:
I think it would be helpful to further divide this into sections that could be used when breaking up a "new model" issue into smaller chunks. Perhaps:

- ### Database Model
- ### Filtering
- ### API Views
- ### UI Views